webui/react/src/pages/WorkspaceList.tsx

```diff
@@ -69,6 +69,8 @@ const WorkspaceList: React.FC = () => {
   const fetchWorkspaces = useCallback(async () => {
     if (!settings) return;
 
+    const abortController = new AbortController();
+    canceler.current = abortController;
```

Comment:
**question:** can't we just consume the useRef directly?
```
canceler.current = new AbortController();

{ signal: canceler.current?.signal }
```

Reply:
intuitively it should be possible, but in practice, I see the `catch` run after the new request sets the canceller, so it causes the page error to be set.